src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -48,16 +49,17 @@ def init_telemetry():
             raise ex
 
 def user_agrees_to_telemetry():
-    # TODO: agreement, needs to take Y/N from the command line
-    # and needs a "skip" param to not show (for scripts)
-    return True
+    return az_config.get('telemetry', 'collect', fallback='n') == 'y'
```

Comment:
This should return a boolean and not 'y'/'n' otherwise, `if not user_agrees_to_telemetry()` doesn't work as intended. It would always return True.